test/RPCSession.cpp

```diff
@@ -107,13 +108,23 @@ string IPCSocket::sendRequest(string const& _req)
 	return string(m_readBuf, m_readBuf + cbRead);
 #else
 	if (send(m_socket, _req.c_str(), _req.length(), 0) != (ssize_t)_req.length())
-		BOOST_FAIL("Writing on IPC failed");
-
-	ssize_t ret = recv(m_socket, m_readBuf, sizeof(m_readBuf), 0);
+		BOOST_FAIL("Writing on IPC failed.");
 
-	// Also consider closed socket an error.
-	if (ret <= 0)
-		BOOST_FAIL("Reading on IPC failed");
+	auto start = chrono::steady_clock::now();
+	ssize_t ret;
+	do
+	{
+		ret = recv(m_socket, m_readBuf, sizeof(m_readBuf), 0);
+		// Also consider closed socket an error.
+		if (ret < 0)
+			BOOST_FAIL("Reading on IPC failed.");
```

Comment:
Negative return value is an error (most likely eth crashed). The idea is just to read again if the result as not yet available.